internal/goroutine/recorder/reader.go

```diff
@@ -0,0 +1,294 @@
+package recorder
+
+import (
+	"encoding/json"
+	"sort"
+	"time"
+
+	"github.com/sourcegraph/sourcegraph/internal/rcache"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// GetBackgroundJobInfos returns information about all known jobs.
+func GetBackgroundJobInfos(c *rcache.Cache, after string, recentRunCount int32, dayCountForStats int32) ([]JobInfo, error) {
```

Comment:
Nitpick: nothing in this file depends on the fact that `c` is an `*rcache.Cache`. It could be any Redis client, right?

```suggestion
type redisClient interface {
  GetHashAll(key string) []string
   // other methods you need in here etc
}

func GetBackgroundJobInfos(c redisClient, after string, recentRunCount int32, dayCountForStats int32) ([]JobInfo, error) {
```

Might make testing easier, loosens dependency between packages, etc.

Reply:
Keegan's broader effort [here](https://github.com/sourcegraph/sourcegraph/pull/46425) should tackle this. WDYT?